### PR TITLE
refactor: work on updating to v12

### PR DESCRIPTION
**Pull request by cfanoulis**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,9 +39,7 @@
 		"standard-version": "^9.0.0",
 		"typescript": "^4.0.2"
 	},
-	"peerDependencies": {
-		"discord.js": "github:discordjs/discord.js#fe7df708e44e0280dfaf0f8e457b154781bb5140"
```
Comment (favna) [inline package.json:43]: Move it back to peer and dev please

**Review by favna — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (cfanoulis) [reply package.json:43]: Resolved in e29dac2

File: typings/index.d.ts [hunk 2]
```diff
@@ -1458,6 +1458,7 @@ declare module 'klasa' {
 
 		export interface DMChannel extends SendAliases, ChannelExtendables {}
 
+		export interface NewsChannel extends SendAliases, ChannelExtendables {}
```
Comment (kyranet) [inline typings/index.d.ts:1461]: Please add some spacing:
```suggestion
		export interface NewsChannel extends SendAliases, ChannelExtendables {}

```

File: src/commands/Admin/enable.js [hunk 3]
```diff
@@ -17,6 +17,6 @@ module.exports = class extends Command {
 				if (String(this.options.shards) !== '${this.client.options.shards}') this.${piece.store}.get('${piece.name}').enable();
 			`);
 		}
-		return message.sendCode('diff', message.language.get('commandEnable', { type: piece.type, name: piece.name }));
+		return message.sendLocale('commandEnable', [piece.type, piece.name], { code: 'diff' });
```
Comment (kyranet) [inline src/commands/Admin/enable.js]: Half-revert this, the arguments must be sent as an object, not as an array of them:
```suggestion
		return message.sendLocale('commandEnable', [{ type: piece.type, name: piece.name }], { code: 'diff' });
```

File: package.json [hunk 4]
```diff
@@ -22,12 +22,11 @@
 	"devDependencies": {
 		"@commitlint/cli": "^9.1.2",
 		"@commitlint/config-conventional": "^9.1.2",
-		"@skyra/eslint-config": "^5.3.1",
-		"@types/node": "^14.6.1",
-		"@typescript-eslint/eslint-plugin": "^3.10.1",
-		"@typescript-eslint/parser": "^3.10.1",
-		"cz-conventional-changelog": "^3.3.0",
-		"discord.js": "github:discordjs/discord.js#fe7df708e44e0280dfaf0f8e457b154781bb5140",
```
Comment (kyranet) [inline package.json:30]: Please restore this as a development dependency, otherwise it won't installed with `yarn`.

**Review by kyranet — CHANGES_REQUESTED** [review]

File: src/commands/Admin/enable.js (hunk 3 above)
Comment (cfanoulis) [reply src/commands/Admin/enable.js]: resolved in 0412b14

File: package.json (hunk 4 above)
Comment (cfanoulis) [reply package.json:30]: resolved in 0412b14

**Review by kyranet — CHANGES_REQUESTED** [review]
Blocking this PR until [settings-gateway](https://github.com/skyra-project/settings-gateway) has been updated for v12's managers.

I'll approve as soon as the PR in SG is done and merged.

**Review by favna — APPROVED** [review]

**Comment by kyranet** [comment]
https://github.com/skyra-project/settings-gateway/pull/178 has landed. Please bump this dependency so I can approve and land this PR.

**Comment by cfanoulis** [comment]
done

**Review by kyranet — APPROVED** [review]
Confirmed hash. https://github.com/skyra-project/settings-gateway/commit/59eb7eb9eea476c8aba09f91040091c7559aebae and [59eb7eb](https://codeload.github.com/skyra-project/settings-gateway/tar.gz/59eb7eb9eea476c8aba09f91040091c7559aebae).

**Pull request merged**